pkg/image/util.go

```diff
@@ -90,29 +90,28 @@ func (u *Utility) Stage(destDir string, imgString string) error {
 
 	if img == nil {
 		registry, err := registries.GetPrivateRegistries(u.findRegistriesYaml())
-
 		if err != nil {
 			return err
 		}
 
-		kcs := []authn.Keychain{registry}
-
 		if _, err := os.Stat(u.imageCredentialProviderConfig); os.IsExist(err) {
 			logrus.Debugf("Image Credential Provider Configuration file %s existed, using plugins from directory %s", u.imageCredentialProviderConfig, u.imageCredentialProviderBinDir)
 			plugins, err := plugin.RegisterCredentialProviderPlugins(u.imageCredentialProviderConfig, u.imageCredentialProviderBinDir)
 			if err != nil {
 				return err
 			}
-			kcs = append(kcs, plugins)
+			registry.DefaultKeychain = plugins
 		} else {
-			kcs = append(kcs, authn.DefaultKeychain)
+			// The kubelet image credential provider plugin also falls back to checking legacy Docker credentials, so only
+			// explicitly set up the go-containerregistry DefaultKeychain if plugins are not configured.
+			// DefaultKeychain tries to read config from the home dir, and will error if HOME isn't set, so also gate on that.
+			if os.Getenv("HOME") != "" {
```

Comment:
Given the explanation above, maybe it's a good idea to at least log a warning if HOME is not set as we currently just silently not setting `DefaultKeychain`.

Reply:
Our desired use case does not have a user directly interfacing with the system-agent, so I am not too "log happy" with it from an interactive perspective